### DISCUSSION TITLE
Log::channel('single')->debug("blubb " . [1,2,3,4]) does NOTHING.

### DISCUSSION THREAD

**Discussion by Martin-Eckleben** [laravel/framework#41979] [Q&A]
- Laravel Version: 9.5.1
- PHP Version: 8.1.4
- Database Driver & Version: pgsql & ?

### Description:

`Log::channel('single')->debug("blubb " . [1,2,3,4]);`
does **NOTHING**.

It does not print the blubb neither the array nor a warning nor an empty line neither does it throw an exception etc.
It does exactly nothing.
I thought my code would never reach that place because of a previous error I'd have probably made :D
This is strongly unexpected and malicious behaviour.

I lost about 3h of debugging rn to this :)

`Log::channel('single')->debug([1,2,3,4]);`
gets prevented by my IDE.

this works:
`Log::channel('single')->debug("blubb " . json_encode([1,2,3,4]));`

Vanilla PHP throws:
`Warning: Array to string conversion in /home/user/scripts/code.php on line 2`

Thanks everybody for your hard great work!

**Comment by anabeto93** [comment]
Hi @Martin-Eckleben your first simple certainly will give you a warning because you  are trying to concatenate an array and a string.

This is not acceptable. `echo "blubb " . [1, 2, 3, 4]`.

Instead of `Log::channel('single')->debug([1,2,3,4]);` you should consider `Log::channel('single')->debug("Some String such as Blubb", [1,2,3,4]);`. You supply a string as the first argument followed by an array of whatever you want to log. If you only care about the array, then you can `json_encode()` to a string before logging it.

**Comment by Martin-Eckleben** [comment]
Hi @anabeto93

I think I didn't get the point across correctly.

> This is not acceptable. echo "blubb " . [1, 2, 3, 4].

Correct.
It throws a PHP Warning and thus I know I did something wrong.

> Log::channel('single')->debug("blubb " . [1,2,3,4]);

Does not throw a warning.
Not even an empty line.
Nothing.
It behaves as if it's never executed.
Why does the Laravel Logging facade suppress vanilla PHP Warnings and does nothing instead?

To explain why that can be important:
I (and many others I think) use it as debugging statement.
If something is printed we know code reached that place.
I spent about 3h debugging the recursive function above my line because I thought the exit condition is wrong and I create an endless loop and therefore nothing is ever logged / happens.
But I found out Laravels logging facade sometimes does simply nothing. 

Thank you for your time and effort.
